lib/SimplifyPointerBitcastPass.cpp

```diff
@@ -44,50 +45,13 @@ clspv::SimplifyPointerBitcastPass::run(Module &M, ModuleAnalysisManager &) {
   return PA;
 }
 
-bool clspv::SimplifyPointerBitcastPass::runOnGEPFromBitcastCstExpr(
-    Module &M) const {
-  SmallVector<GetElementPtrInst *, 16> WorkList;
+bool clspv::SimplifyPointerBitcastPass::runOnInstFromCstExpr(Module &M) const {
+  bool Changed = false;
   for (Function &F : M) {
-    for (BasicBlock &BB : F) {
-      for (Instruction &I : BB) {
-        // If we have a GEP instruction...
-        if (auto GEP = dyn_cast<GetElementPtrInst>(&I)) {
-          // ... whose source is a bitcast constant expression
-          if (const auto CE =
-                  dyn_cast<ConstantExpr>(GEP->getPointerOperand())) {
-            if (CE->getOpcode() == Instruction::BitCast) {
-              // ... whose source pointer element is not a struct:
-              //
-              // This case happen in some particular scenario, where it is not
-              // needed to simplify it. Mostly when using global array
-              // variables. In the end llvm managed to deal with it without us
-              // having to simplify it. Trying to simplify would make it very
-              // complicated for the ReplacePointerBitcast pass.
-              if (!CE->getOperand(0)
-                       ->getType()
-                       ->getNonOpaquePointerElementType()
-                       ->isStructTy()) {
-                // ... record the GEP as something we need to process.
-                WorkList.push_back(GEP);
-              }
-            }
-          }
-        }
-      }
-    }
+    Changed |= BitcastUtils::RemovedCstExprFromFunction(&F);
```

Comment:
Refactor this to the run method when changing the function to hit every constantexpr.

Reply:
Not sure what you expected here